experimental/FTheoryTools/src/WeierstrassModels/attributes.jl

```diff
@@ -242,6 +242,33 @@ julia> length(singular_loci(w))
 2
 ```
 """
+@attr Vector{<:Tuple{<:MPolyIdeal{<:MPolyRingElem}, Tuple{Int64, Int64, Int64}, String}} function singular_loci(w::WeierstrassModel)
+  @req (base_space(w) isa NormalToricVariety || base_space(w) isa FamilyOfSpaces) "Singular loci of Weierstrass model is currently only supported for toric varieties and families of spaces as base space"
+  B = irrelevant_ideal(base_space(w))
+  d_primes = factor(discriminant(w))
+  nontrivial_d_primes = [k for k in d_primes if _is_nontrivial(ideal([k[1]]), B)]
+  f_primes = factor(weierstrass_section_f(w))
+  g_primes = factor(weierstrass_section_g(w))
+  f_prime_dict = Dict(fp[1] => fp[2] for fp in f_primes)
+  g_prime_dict = Dict(gp[1] => gp[2] for gp in g_primes)
```

Comment:
Just to say (and feel free to ignore this) you can also write this
```suggestion
  f_primes = factor(weierstrass_section_f(w))
  g_primes = factor(weierstrass_section_g(w))
  f_prime_dict = Dict(p => n for (p,n) in f_primes)
  g_prime_dict = Dict(p => n for (p,n) in g_primes)
```
or even this
```suggestion
  f_prime_dict = Dict(p => n for (p,n) in factor(weierstrass_section_f(w)))
  g_prime_dict = Dict(p => n for (p,n) in factor(weierstrass_section_g(w)))
```

Of course even simpler is
```suggestion
  f_prime_dict = factor(weierstrass_section_f(w)).fac
  g_prime_dict = factor(weierstrass_section_g(w)).fac
```
but that is using internals. I think what we really should do is change the `getindex` method for `Fac` to return 0 for missing keys instead of raising an error (as suggested in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1959), then you could just do this:
```suggestion
  f_primes = factor(weierstrass_section_f(w))
  g_primes = factor(weierstrass_section_g(w))
```
and replace `get(f_prime_dict, d_prime[1], 0)` by `f_prime[d_prime[1]]`.